stl/inc/vector

```diff
@@ -1175,11 +1167,20 @@ public:
             _Clear_and_reserve_geometric(_Newsize);
             if constexpr (_Nothrow_construct) {
                 _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
-                _ASAN_VECTOR_CREATE;
             } else {
-                _ASAN_VECTOR_GUARD_CREATE;
+#ifdef _INSERT_VECTOR_ANNOTATION
+                _TRY_BEGIN
+#endif // _INSERT_VECTOR_ANNOTATION
                 _Mylast = _Uninitialized_fill_n(_Myfirst, _Newsize, _Val, _Al);
+#ifdef _INSERT_VECTOR_ANNOTATION
+                _CATCH_ALL
+                _ASAN_VECTOR_CREATE;
+                _RERAISE;
+                _CATCH_END
+#endif // _INSERT_VECTOR_ANNOTATION
             }
+
+            _ASAN_VECTOR_CREATE;

```

Comment:
It would be a substantial improvement to replace:
```c++
if constexpr (condition) {
  meow();
} else {
  _TRY_BEGIN
  woof();
  _CATCH_ALL
  _ASAN_VECTOR_CREATE;
  _RERAISE;
  _CATCH_END
}
_ASAN_VECTOR_CREATE;
```
 with an RAII guard that unconditionally calls `_ASAN_VECTOR_CREATE` in its destructor:
```c++
_ASAN_MEOW(*this);
if constexpr (condition) {
  meow();
} else {
  woof();
}
```

Reply:
I would second that this is quite an improvement. I am a bit unclear how to properly implement this but an improvement indeed.

Note that we already have the "revert everything in case of exception" usecase for the one RAII guard we have. That said we should be able to simply call the appropriate `_ASAN_VECTOR_CREATE` in the destructor of that second type. AFAIK we can get away with the destructor in both cases

Reply:
Before rewriting with the try/catch I was essentially highjacking the existing annotator's destructor to get the effect we wanted.

Reply:
I would rather not given that we do not need to hold any state